commands/azureCommands/acr-build-logs.ts

```diff
@@ -0,0 +1,381 @@
+                                <div class = 'button-holder'>
+                                    <button id= "log${i}" class="viewLog">Open Logs</button>
+                                </div>
+                        </div>
+                    </div>`
+        });
+    }
+    if (startItem) {
+        panel.webview.postMessage({ 'type': 'endContinued' });
+    } else {
+        panel.webview.postMessage({ 'type': 'end' });
+    }
+}
+/** Create the table in which to push the build logs */
+function getWebviewContent(scriptFile: vscode.Uri, stylesheet: vscode.Uri): string {
+    return `<!DOCTYPE html>
+    <html lang="en">
+    <head>
+        <link rel="stylesheet" type="text/css" href="${stylesheet}">
+        <meta charset="UTF-8">
+        <meta http-equiv="Content-Security-Policy" content="frame-src vscode-resource:; img-src vscode-resource: https:; script-src vscode-resource:; style-src vscode-resource:;">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <title>Logs</title>
+    </head>
+
+    <body>
+        <div id = "header">
+            <table id="headerTable">
+                <th class = 'widthControl'>Build Name </th>
+                <th class = 'widthControl'>BuildTask </th>
+                <th class = 'widthControl'>Status </th>
+                <th class = 'widthControl'>Start Time </th>
+                <th class = 'widthControl'>Finish Time </th>
+                <th class = 'widthControl'>Platform </th>
+                <td></td>
+            </table>
+        </div>
+        <div id = 'core'>
+        </div>
+        <div class = 'loadMoreBtn'>
+            <button id= "loadBtn" class="viewLog">Load More Logs</button>
+        </div>
+        <script src= "${scriptFile}"></script>
+    </body>
+`;
+}
+/** Setup communication with the webview sorting out received mesages from its javascript file */
+function setupCommunication(panel: vscode.WebviewPanel, logData: LogData): void {
+    panel.webview.onDidReceiveMessage(async (message) => {
+        if (message.logRequest) {
+            const itemNumber: number = +message.logRequest.id;
+            logData.getLink(itemNumber).then((url) => {
+                if (url !== 'requesting') {
+                    openLog(url, logData.logs[itemNumber].buildId);
+                }
+            })
+        } else if (message.loadMore) {
+            try {
+                await logData.loadMoreLogs();
+            } catch (error) {
+                vscode.window.showErrorMessage(error.message);
+            }
+
+            addLogsToWebView(panel, logData, logData.logs.length);
+        }
+    });
+}
+
+//# LOGS
+/** Passes the log text into a document provider */
+function createLogView(text: string, title: string): void {
+    const scheme = 'purejs';
+    let uri: vscode.Uri;
+    try {
+        let query = JSON.stringify({ 'log': makeBase64(text) });
+        uri = vscode.Uri.parse(`${scheme}://authority/${title}?${query}#idk`);
```

Comment:
>title [](start = 56, length = 5)

So you download the log and pass it as part of url? How large of the log based64 string you can pass in the url?